src/core/utils/CustomMulticallJsonRpcProvider.ts

```diff
@@ -1,33 +1,31 @@
-import { providers } from 'ethers'
+import { JsonRpcProvider, Networkish, TransactionRequest } from 'ethers'
 import { logger } from './logger'
 import { MulticallQueue } from './multicall'
 
-export class CustomMulticallJsonRpcProvider extends providers.StaticJsonRpcProvider {
+export class CustomMulticallJsonRpcProvider extends JsonRpcProvider {
   private multicallQueue: MulticallQueue
   constructor({
     url,
     network,
     multicallQueue,
   }: {
     url: string
-    network: providers.Networkish
+    network: Networkish
     multicallQueue: MulticallQueue
   }) {
     super(url, network)
     this.multicallQueue = multicallQueue
   }
 
-  async call(
-    transaction: providers.TransactionRequest,
-    blockTag?: providers.BlockTag | Promise<providers.BlockTag>,
-  ): Promise<string> {
-    if (blockTag) {
-      logger.debug('Intercepted eth_call, using multicall queue implementation')
-      return super.call(transaction, blockTag)
+  async call(transaction: TransactionRequest): Promise<string> {
+    logger.debug(
+      transaction,
+      'Intercepted eth_call, using multicall queue implementation',
+    )
+    if (transaction.blockTag) {
+      return super.call(transaction)
```

Comment:
`blockTag` is now part of the `TransactionRequest`, they split it out internally within the provider.

Reply:
oh thats cool are there any overrides now?

Reply:
These are all the overrides. Before, `blockTag` was an additional optional parameter. Much better this way IMO.

https://github.com/ethers-io/ethers.js/blob/main/src.ts/providers/provider.ts#L124